scipy/special/basic.py

```diff
@@ -2229,29 +2229,58 @@ def perm(N, k, exact=False):
         return vals
 
 
+# http://stackoverflow.com/a/16327037/125507
+def _range_prod(lo, hi):
+    """
+    Product of a range of numbers
+
+    Returns the product of
+    lo * (lo+1) * (lo+2) * ... * (hi-2) * (hi-1) * hi
+    = hi! / (lo-1)!
+    """
+    if lo + 1 < hi:
+        mid = (hi + lo) // 2
+        return _range_prod(lo, mid) * _range_prod(mid + 1, hi)
+    if lo == hi:
+        return lo
+    return lo * hi
+
+
 def factorial(n, exact=False):
-    """The factorial function, n! = special.gamma(n+1).
+    """
+    The factorial of a number or array of numbers
 
-    If exact is 0, then floating point precision is used, otherwise
-    exact long integer is computed.
+    The factorial of non-negative integer `n` is the product of all
+    positive integers less than or equal to `n`:
 
-    - Array argument accepted only for exact=False case.
-    - If n<0, the return value is 0.
+    n! = n * (n - 1) * (n - 2) * ... * 1
```

Comment:
Base asterisks are going to confuse Sphinx. I *think* you want to use `::` at the end of the previous paragraph and indent the formula (untested --- it's hard to be sure with sphinx).